### PR TITLE
feat: dropdown grouping added in routing

### PR DESCRIPTION
**Pull request by PritishBudhiraja**
## Type of Change

- [ ] Bugfix
- [x] New feature
- [ ] Enhancement
- [ ] Refactoring
- [ ] Dependency updates
- [ ] Documentation
- [ ] CI/CD

## Description

Screenshot 2023-12-15 at 3 48 46 PM

## Checklist

- [x] I ran `npm run re:build`
- [x] I reviewed submitted code
- [ ] I added unit tests for my changes where possible

### CODE REVIEW

File: src/screens/HyperSwitch/RoutingRevamp/AdvancedRoutingUIUtils.res [hunk 1]
```diff
@@ -283,11 +283,39 @@ let metaInput = (id, keyType) =>
...
 module FieldInp = {
   @react.component
-  let make = (~ops, ~prefix, ~onChangeMethod) => {
+  let make = (~prefix, ~onChangeMethod) => {
     let field = ReactFinalForm.useField(`${prefix}.lhs`).input
     let op = ReactFinalForm.useField(`${prefix}.comparison`).input
     let val = ReactFinalForm.useField(`${prefix}.value.value`).input
 
+    let keyDescriptionMapper = Window.getDescriptionCategory()->MapTypes.changeType
+    let convertedValue = LogicUtils.convertMapObjectToDict(keyDescriptionMapper)
```
Comment (JeevaRamu0104) [inline src/screens/HyperSwitch/RoutingRevamp/AdvancedRoutingUIUtils.res:292]: can we encapsulate the `keyDescriptionMapper` and `convertedValue` under a memo function and return a single value from the memo

File: src/screens/HyperSwitch/RoutingRevamp/AdvancedRoutingUIUtils.res [hunk 2]
```diff
@@ -283,11 +283,39 @@ let metaInput = (id, keyType) =>
...
+    let options = convertedValue->Js.Dict.keys->Js.Array2.reduce((acc, ele) => {
```
Comment (JeevaRamu0104) [inline src/screens/HyperSwitch/RoutingRevamp/AdvancedRoutingUIUtils.res:293]: same for the options can we use memo

Comment (PritishBudhiraja) [reply src/screens/HyperSwitch/RoutingRevamp/AdvancedRoutingUIUtils.res]: updated

File: src/screens/HyperSwitch/RoutingRevamp/AdvancedRoutingUIUtils.res (hunk 1 above)
Comment (PritishBudhiraja) [reply src/screens/HyperSwitch/RoutingRevamp/AdvancedRoutingUIUtils.res]: updated

File: src/screens/HyperSwitch/RoutingRevamp/AdvancedRoutingUIUtils.res [hunk 3]
```diff
@@ -283,10 +283,50 @@ let metaInput = (id, keyType) =>
...
+    let convertedValue = React.useMemo1(() => {
+      let keyDescriptionMapper = descriptionCategory->MapTypes.changeType
+      keyDescriptionMapper->LogicUtils.convertMapObjectToDict
+    }, [descriptionCategory])
+
+    let options = React.useMemo1(
+      () => convertedValue->Js.Dict.keys->Js.Array2.reduce((acc, ele) => {
+          open LogicUtils
+          convertedValue
+          ->Js.Dict.get(ele)
```
Comment (Riddhiagrawal001) [inline src/screens/HyperSwitch/RoutingRevamp/AdvancedRoutingUIUtils.res:301]: Please use getOptionalArrayFromDict from LogicUtils

File: src/screens/HyperSwitch/RoutingRevamp/AdvancedRoutingUIUtils.res [hunk 4]
```diff
@@ -283,10 +283,50 @@ let metaInput = (id, keyType) =>
...
+          ->Belt.Option.getWithDefault([]->Js.Json.array)
+          ->Js.Json.decodeArray
+          ->Belt.Option.getWithDefault([])
+          ->Js.Array2.forEach(
+            value => {
+              let dictValue = value->LogicUtils.getDictFromJsonObject
+              let extractValueFromDict = key =>
+                dictValue
+                ->Js.Dict.get(key)
+                ->Option.getWithDefault(""->Js.Json.string)
```
Comment (Riddhiagrawal001) [inline src/screens/HyperSwitch/RoutingRevamp/AdvancedRoutingUIUtils.res:311]: Please use getString from LogicUtils

File: src/screens/HyperSwitch/RoutingRevamp/AdvancedRoutingUIUtils.res (hunk 3 above)
Comment (PritishBudhiraja) [reply src/screens/HyperSwitch/RoutingRevamp/AdvancedRoutingUIUtils.res]: updated

File: src/screens/HyperSwitch/RoutingRevamp/AdvancedRoutingUIUtils.res (hunk 4 above)
Comment (PritishBudhiraja) [reply src/screens/HyperSwitch/RoutingRevamp/AdvancedRoutingUIUtils.res]: updated

**Review by JeevaRamu0104 — APPROVED** [review]

**Review by Riddhiagrawal001 — APPROVED** [review]

**Pull request merged**